### ISSUE TITLE
!todo bordure of impaled quarter is not bordering edge

### ISSUE THREAD

**Issue by karlwilcox** [drawshield/Drawshield-Code#176]
!todo bordure of impaled quarter is not bordering edge
!draw Quarterly
I or a lion sable armed and langued gules impaled with gules a bordure sable
II azure
III gules
IV vert

**Comment by ericscheid** [comment]
> [astrid — Today at 1:08 AM](https://discord.com/channels/775325424854630440/775347423597297704/852927307353554951)
I put the colors as placeholders but seems to be a rendering glitch with the first quarter
I tried to impale the gules half onto the or half and the bordure on the gules half didn’t work

![image](https://user-images.githubusercontent.com/2605467/121716672-a48e1180-cb23-11eb-9bb4-3d2619032a30.png)

**Comment by karlwilcox** [comment]
Kind of obscure and also *very* hard, it requires a different bordure edge depending on which quarter, and which side the impaled bordure is in.

**Comment by ericscheid** [comment]
Yep, and then also whether the field is quarted into 9ths or 6ths too.

Most ppl have a mental model of the edge being traced and indented dynamically, while the code has a working model of applying a pre-defined shape. The mental model method is no easier, given weird shapes like the swiss shield (where the outer edge is different from the inner edge)

Could be the way though, as will also handle `azure, a chief argent, a bordure or, drawn in a german shape`

----

I might run up a spreadsheet of combinations, look for an algo with fewest special cases.
